agent/src/scheduler.cc

```diff
@@ -214,28 +212,38 @@ void scheduler::update(const engine_to_agent_request_ptr& conf) {
     }
 
     // first we group checks by check_interval
-    std::map<uint32_t, std::vector<const Service*>> group_serv;
-    for (const auto& serv : conf->config().services()) {
-      uint32_t check_interval = serv.check_interval();
-      if (check_interval == 0) {
-        check_interval = 60;  // one minute by default
+    std::map<uint32_t, std::vector<Service*>> group_serv;
+    for (auto& serv : *conf->mutable_config()->mutable_services()) {
+      if (serv.check_interval() == 0) {
+        serv.set_check_interval(60);  // one minute by default
+      }
+      if (serv.retry_interval() == 0) {
+        serv.set_retry_interval(60);  // one minute by default
       }
-      group_serv[check_interval].push_back(&serv);
+      if (serv.max_attempts() == 0) {
+        serv.set_max_attempts(1);  // one attempt by default
+      }
+      uint32_t check_interval = serv.check_interval();
+      uint32_t retry_interval = serv.retry_interval();
+      auto min_interval = std::min(check_interval, retry_interval);
+      group_serv[min_interval].push_back(&serv);
```

Comment:
I'm not sure, but I think it would be better
```suggestion
      group_serv[check_interval].push_back(&serv);
```